cardano-api/src/Cardano/Api/Convenience/Query.hs

```diff
@@ -90,13 +96,16 @@ queryStateForBalancedTx era networkId allTxIns = runExceptT $ do
       stakePoolsQuery = QueryInEra qeInMode . QueryInShelleyBasedEra qSbe $ QueryStakePools
 
   -- Query execution
-  utxo <- ExceptT $ executeQueryCardanoMode era networkId utxoQuery
-  pparams <- ExceptT $ executeQueryCardanoMode era networkId pparamsQuery
-  eraHistory <- firstExceptT AcqFailure $ ExceptT $ queryNodeLocalState localNodeConnInfo Nothing eraHistoryQuery
-  systemStart <- firstExceptT AcqFailure $ ExceptT $ queryNodeLocalState localNodeConnInfo Nothing systemStartQuery
-  stakePools <- ExceptT $ executeQueryCardanoMode era networkId stakePoolsQuery
-
-  return (utxo, pparams, eraHistory, systemStart, stakePools)
+  ( lift $ executeLocalStateQueryExpr localNodeConnInfo Nothing $ runExceptT $ runExceptT $ do
+      utxo <- ExceptT $ ExceptT $ queryExpr utxoQuery
+      pparams <- ExceptT $ ExceptT $ queryExpr pparamsQuery
+      eraHistory <- ExceptT $ fmap Right $ ExceptT $ queryExpr eraHistoryQuery
+      systemStart <- ExceptT $ fmap Right $ ExceptT $ queryExpr systemStartQuery
+      stakePools <- ExceptT $ ExceptT $ queryExpr stakePoolsQuery
+      pure (utxo, pparams, eraHistory, systemStart, stakePools)
+    ) & onLeft (throwE . AcqFailure)
+      & onLeft (throwE . ConvenienceUnsupportedNtcVersionError)
+      & onLeft (throwE . QueryEraMismatch)
```

Comment:
Some queries like `QueryUTxO`, `QueryProtocolParameters` and `QueryStakePools` return their own errors so needs to be handled with an extra `ExceptT` stack.